### PR TITLE
Loosen Dict->DataFrame conversion to Associative->DataFrame

### PR DESCRIPTION
**Pull request by kmsquire**
Trivial fix for #950.

### CODE REVIEW

**Comment by femtotrader** [comment]
hum....

I see `sort!(dnames)` ... so I think it will reorder columns (and not preserve columns orders) !

**Comment by kmsquire** [comment]
Whoops!  I should pay more attention.  I can add another method which just uses the iteration order...

**Comment by kmsquire** [comment]
Okay, this should be work better.

A better solution, yet, would probably be to have a constructor which had a `sort={true/false}` keyword argument, which would allow any `Associative` to work in a reasonable way.  As it is with this PR, `Dicts` will be sorted, and everything else will maintain its order, but, e.g., other `Associative` types defined outside of `Base` (and `DataStructures.jl` might also benefit from sorting.

**Comment by nalimilan** [comment]
> A better solution, yet, would probably be to have a constructor which had a sort={true/false} keyword argument, which would allow any Associative to work in a reasonable way. As it is with this PR, Dicts will be sorted, and everything else will maintain its order, but, e.g., other Associative types defined outside of Base (and DataStructures.jl might also benefit from sorting.

I guess we can wait for a future `OrderedAssociative` trait for that. For now, I'd say your approach is fine.

Could you add a test for each case though?

**Comment by kmsquire** [comment]
There was already a test for the first case (`Dict`, columns are sorted).  I added a test for the second (`Associative`, column order same as key order).

File: test/conversions.jl [hunk 1]
```diff
@@ -55,6 +55,13 @@ module TestConversions
     @test df[:b] == b
     @test df[:c] == c
 
+    df = invoke(convert, (Type{DataFrame}, Associative), DataFrame, di)
```
Comment (nalimilan) [inline test/conversions.jl]: Why not test the actual use the case this PR fixes, and use an `OrderedDict`? You can add DataStructures.jl to `test/REQUIRES`, it's not a terribly heavy dependency.

Comment (kmsquire) [inline test/conversions.jl]: I thought about that. This version seemed easy enough, but I have no problem adding a test targeting OrderedDicts.

**Comment by kmsquire** [comment]
I added tests for conversion from `OrderedDicts` and `SortedDicts`.

**Comment by nalimilan** [comment]
Great, thanks! Though (sorry for being such a pain) these tests are not enough to test that we respect the actual order of elements, since they are lexicographically sorted already. How about changing the order to e.g. b, c, a? Then I promise I'll merge! :-)

**Comment by kmsquire** [comment]
:-) No worries, @nalimilan!  Done.

**Comment by nalimilan** [comment]
Perfect, thanks!

**Pull request merged**